examples/config.c

```diff
@@ -0,0 +1,304 @@
+#ifndef _POSIX_C_SOURCE
+#define _POSIX_C_SOURCE 200809L
+#endif
+#include <stdlib.h>
+#include <limits.h>
+#include <getopt.h>
+#include <string.h>
+#include <unistd.h>
+#include <wlr/util/log.h>
+#include <wlr/types/wlr_geometry.h>
+#include "shared.h"
+#include "config.h"
+#include "ini.h"
+
+static void usage(const char *name, int ret) {
+	fprintf(stderr,
+		"usage: %s [-C <FILE>]\n"
+		"\n"
+		" -C <FILE>      Path to the configuration file (default: wlr-example.ini).\n"
+		"                See `examples/wlr-example.ini.example` for config file documentation.\n", name);
+
+	exit(ret);
+}
+
+static struct wlr_geometry *parse_geometry(const char *str) {
+	// format: {width}x{height}+{x}+{y}
+	if (strlen(str) > 255l) {
+		wlr_log(L_ERROR, "cannot parse geometry string, too long");
+		return NULL;
+	}
+
+	char *buf = strdup(str);
+	struct wlr_geometry *geo = calloc(1, sizeof(struct wlr_geometry));
+
+	bool has_width = false;
+	bool has_height = false;
+	bool has_x = false;
+	bool has_y = false;
+
+	char *pch = strtok(buf, "x+");
+	while (pch != NULL) {
+		errno = 0;
+		char *endptr;
+		long val = strtol(pch, &endptr, 0);
+
+		if ((errno == ERANGE && (val == LONG_MAX || val == LONG_MIN))
+				|| (errno != 0 && val == 0)) {
+			goto invalid_input;
+		}
+
+		if (endptr == pch) {
+			goto invalid_input;
+		}
+
+		if (!has_width) {
+			geo->width = val;
+			has_width = true;
+		} else if (!has_height) {
+			geo->height = val;
+			has_height = true;
+		} else if (!has_x) {
+			geo->x = val;
+			has_x = true;
+		} else if (!has_y) {
+			geo->y = val;
+			has_y = true;
+		} else {
+			break;
+		}
+		pch = strtok(NULL, "x+");
+	}
+
+	if (!has_width || !has_height || !has_x || !has_y) {
```

Comment:
+x+y could be assumed to be 0 if omitted?

Reply:
Yeah that's probably a good idea.